mdl/src/main/cft/mdlCreateSecurityGroups.yml

```diff
@@ -330,6 +330,108 @@ Resources:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: arn:aws:logs:*:*:*
+              - Effect: Allow
+                Action:
+                  - ec2:DescribeSecurityGroups
+                  - ec2:RevokeSecurityGroupEgress
+                  - ec2:RevokeSecurityGroupIngress
+                  - ec2:DeleteSecurityGroup
+                Resource: '*'
+  DeleteSecurityGroupLambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: 'index.handler'
+      Code:
+        ZipFile: |
+          import cfnresponse
+          import boto3
+
+          def handler(event, context):
+              vpc_id = event['ResourceProperties']['VpcId']
+              physical_id = "DeleteEmrDefaultSecurityGroupLambdaPhysicalID"
+              response_data = {}
+              if event['RequestType'] == 'Delete':
+                  emr_sg_default = [
+                      'ElasticMapReduce-Master-Private',
+                      'ElasticMapReduce-ServiceAccess',
+                      'ElasticMapReduce-Slave-Private'
+                  ]
+                  client = boto3.client('ec2')
+                  try:
+                      response = client.describe_security_groups(
+                          Filters=[
+                              {
+                                  'Name': 'vpc-id',
+                                  'Values': [
+                                      vpc_id,
+                                  ]
+                              },
+                              {
+                                  'Name': 'group-name',
+                                  'Values': emr_sg_default
+                              },
+                          ]
+                      )
+                      sgs = response['SecurityGroups']
+                      for item in sgs:
+                          print('{}:{}'.format(item['GroupName'], item['GroupId']))
+                          empty_ingress_egress( item['GroupId'])
+
+                      for item in sgs:
+                          client.delete_security_group(
+                              GroupId=item['GroupId']
+                          )
+                  except Exception as e:
+                      response_data['Message'] = 'Unexpected error. Exception: {}'.format(e)
+                      cfnresponse.send(event, context, cfnresponse.FAILED, response_data, physical_id)
+                  response_data['Message'] = 'Successfully Deleted {} EMR default SGs'.format(len(sgs))
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data, physical_id)
+              else:
+                  response_data['Message'] = 'Nothing to do.'
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data, physical_id)
+
+          def empty_ingress_egress( group_id):
```

Comment:
`revoke_ingress_egress`?

Reply:
yeah, the method is calling method ingress() and egress()